src/model/operation/transform.js

```diff
@@ -187,9 +203,160 @@ function initializeContext( opsA, opsB, options ) {
 		return this.document.history.isUndoneOperation( originalOp );
 	};
 
+	context.getRelation = function( opA, opB ) {
+		if ( !options.useContext ) {
+			return null;
+		}
+
+		const origB = this.originalOperations.get( opB );
+		const undoneB = this.document.history.getUndoneOperation( origB );
```

Comment:
Undone operation? Comment is needed.

Reply:
I am not sure what exactly should be commented here, but I'll add extensive comments in a followup to explain every of those algorithms.

Reply:
Nevermind, it was late. ;)